app/routes/session.py

```diff
@@ -128,9 +151,76 @@ def login() -> Response:
 
     cookie_session["language_code"] = metadata.language_code
 
+    if (dataset_id := metadata["sds_dataset_id"]) and ru_ref:
+        get_supplementary_data(
+            supplementary_data_url=SUPPLEMENTARY_DATA_URL,
```

Comment:
survey_id from the json isn't being passed into the validation and isn't being validated `test_supplementary_data` has survey_id 0 and the mock tests are using survey ids of `001` and `002` the ticket states that this should be invalid - so I think another validation method might be needed in the marshmallow schemas?

Reply:
This has been updated now 👍